var/spack/repos/builtin/packages/mkl/package.py

```diff
@@ -71,4 +71,17 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         spack_env.set('MKLROOT', self.prefix)
 
     def setup_environment(self, spack_env, env):
+        # Remove paths that were guessed but are incorrect for this package.
+        env.remove_path('LIBRARY_PATH',
```

Comment:
AFAIK Spack does not use `LD_LIBRARY_PATH` or alike anywhere, everything is RPATH'ed, why do you need all those changes?